kivymd/uix/navigationdrawer.py

```diff
@@ -696,9 +696,14 @@ def on_touch_move(self, touch):
                 self.status = "closing_with_swipe"
 
         if self.status in ("opening_with_swipe", "closing_with_swipe"):
-            self.open_progress = max(
-                min(self.open_progress + touch.dx / self.width, 1), 0
-            )
+            if self.anchor == "left":
+                self.open_progress = max(
+                    min(self.open_progress + touch.dx / self.width, 1), 0
+                )
+            else:
+                self.open_progress = max(
+                    min(self.open_progress - touch.dx / self.width, 1), 0
+                )
```

Comment:
The fix works but,
it's more efficient it you only change the value that's affecting the code, in this case
instead of doing
```py
 if self.anchor == "left":
                self.open_progress = max(
                    min(self.open_progress + touch.dx / self.width, 1), 0
                )
            else:
                self.open_progress = max(
                    min(self.open_progress - touch.dx / self.width, 1), 0
                )
```
i would say the code should be:
```py
self.open_progress = max(
                min(
                    self.open_progress
                    + (touch.dx if self.anchor == "left" else -touch.dx)
                    / self.width,
                    1,
                ),
                0,
            )
```

Reply:
Next time ill post some test code for the solution. I didn't know this "interpreter works faster with inline if else clauses, since the whole if else blocks takes a little more analysis time." I will for sure make the improvement mentioned abouve and re-push those changes.